src/services/identity/DeploymentsService.ts

```diff
@@ -115,6 +118,83 @@ class DeploymentsService {
           .map(() => amplifyInfo)
       })
   }
+
+  getDeploymentInfoFromSiteId = async (repoId: string) => {
+    const deploymentInfo = await this.repository.findOne({
+      where: {
+        siteId: repoId,
+      },
+    })
+    if (!deploymentInfo)
+      return errAsync(new NotFoundError("Site has not been deployed!"))
+    return okAsync(deploymentInfo)
+  }
+
+  updateAmplifyPassword = async (
+    repoName: string,
+    encryptedPassword: string,
+    iv: string,
+    enablePassword: boolean
+  ) => {
+    const deploymentInfo = await this.repository.findOne({
+      include: [
+        {
+          model: Site,
+          required: true,
+          include: [
+            {
+              model: Repo,
+              required: true,
+              where: {
+                name: repoName,
+              },
+            },
+          ],
+        },
+      ],
+    })
+    if (!deploymentInfo)
+      return errAsync(`Deployment for ${repoName} does not exist`)
+    const { id, hostingId: appId } = deploymentInfo
+    let updateAppInput
+    if (!enablePassword) {
```

Comment:
could we just branch here and hide the rest of the logic into 2 separate methods? 

very confusing to figure out where/what is used, especially because the purpose of the two is diametrically opposed (1 sets password, 1 removes)

Reply:
Yeah that makes sense, https://github.com/isomerpages/isomercms-backend/pull/806/commits/f13e3f214b5fcc692fdc2f8d486959f52d0c2fa5